### ISSUE TITLE
2.10 does not prompt for cfgupgrade

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#839]
Originally reported of Google Code with ID 783.

```
<b>What steps will reproduce the problem?</b>
1. Upgrade a cluster from 2.9 to 2.10
2. Try starting masterd
3. Masterd fails but does not inform the user that a configuration upgrade is
   necessary.

<b>What is the expected output? What do you see instead?</b>

The usual error message is: 

 The current Ganeti softare expect version %s, but the on-disk configuration
 file has version %s. [...]".

What actually appears during an upgrade to 2.10 is:

 Configuration error while opening the configuration file: Object Disk doesn't
 support the parameter 'physical_id' This might be caused by an incomplete
 software upgrade or by a corrupted configuration file. Until the problem is
 fixed the master daemon cannot start.

This seems to come from CheckMasterd(), where opening the config raises a
ConfigurationError instead of a ConfigVersionMismatch. The problem arises from
the fact that calling ConfigData.FromDict() in ConfigWriter._OpenConfig() fails
(even before ConfigWriter._UpgradeConfig() has a chance to run) because the
Disk objects have no "physical_id" slot. A suggested fix would be to either
keep the slot in the class (but not use it), or remove the field from the
config after serializer.Load() but before objects.ConfigData.FromDict().

```
Originally added on 2014-03-31 07:27:15 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-03-31 08:46:05 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 2dc0acb98f2ca54b3047f66d720ddc21c93d5ea4
Author: Klaus Aehlig <aehlig@google.com>
Date:   Mon Mar 31 14:11:04 2014 +0200

    Verify configuration version number before parsing
    
    As the attempt to convert the dict used as json representation
    of the configuration into a configuration object already makes
    assumptions about the internal representation, verify the version
    before such an attempt. Fixes issue 783.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Hrvoje Ribicic <riba@google.com>

```
Originally added on 2014-03-31 13:42:58 +0000 UTC.

Mentioned Issue with GoogleCode ID 783 has been migrated to GitHub with ID #839.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released in 2.10.3.
```
Originally added on 2014-04-16 10:37:20 +0000 UTC.

Changed State: Released
